src/src/com/microsoft/aad/adal/HttpWebRequest.java

```diff
@@ -93,7 +94,7 @@ private HttpURLConnection setupConnection() throws IOException {
             throw new IllegalArgumentException("requestURL");
         }
         HttpURLConnection.setFollowRedirects(true);
-        final HttpURLConnection connection = (HttpURLConnection)mUrl.openConnection();
+        final HttpURLConnection connection = HttpUrlConnectionFactory.createHttpUrlConnection(mUrl);
```

Comment:
murl is private, then how i inject the dependency in it?

---
In reply to: [63405654](https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/611#discussion_r63405654) [](ancestors = 63405654)